src/lazy/expanded/macro_table.rs

```diff
@@ -511,18 +528,21 @@ impl MacroTable {
         Ok(id)
     }
 
+    pub(crate) fn append_macro(&mut self, macro_ref: &Arc<Macro>) -> IonResult<()> {
```

Comment:
🪧 This method was extracted from the `append_all_macros_from` function below. It is now also be used in `compile_system_macros()`.